전남대_18조_Garden_5주차

chaemina opened the pull request:
## Summary

전남대_18조_Garden_5주차 (dev->master)

## Details

[레이아웃 구현 PR](https://github.com/Step3-kakao-tech-campus/Team18_FE/pull/16#issue-1929294445) 

[채팅 리스트 창 컴포넌트 구현 PR](https://github.com/Step3-kakao-tech-campus/Team18_FE/pull/20#issue-1931393890) 

[로그인 및 회원가입 페이지 폼 구현 ](https://github.com/Step3-kakao-tech-campus/Team18_FE/pull/19#issue-1931368161) 

## Issue

- Related #3 
- Related #18 
- Related #14 

## ETC

- 그 외의 오류 해결 및 4주차 코드 리뷰 바탕으로 코드를 수정함

bada308 committed:
fix: prop-types 제거

prop-types 라이브러리 제거함
"react/prop-types" 규칙 off로 설정함

bada308 committed:
Merge branch 'master' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/4-branch-chatting

jsh1147 committed:
Merge pull request #12 from Step3-kakao-tech-campus/feature/4-branch-chatting

prop-types & reset.css 제거

jsh1147 committed:
fix: CSS 미적용 문제 수정

tailwindCSS의 적용을 담당하는 App.css가
App.jsx에서 import되지 않고 있던 문제를 수정함
추가로, 사용되지 않는 index.css를 제거함

jsh1147 committed:
Merge pull request #13 from jsh1147/feat/mentoring

CSS 미적용 문제 수정

chaemina committed:
fix : tailwind css color 수정 및 추가

title color 수정, paragraph color 추가

jsh1147 committed:
fix: 불필요 .gitkeep 제거

디렉터리 내에 파일이 생겨 더이상 불필요해진
.gitkeep 파일을 제거함

jsh1147 committed:
feat: 레이아웃 구조 구성

레이아웃(GNB, 컨텐츠, footer) 구조를 적용함
컨텐츠를 감싸는 main 태그 영역에 배경색을 입힘

Related #3

jsh1147 committed:
feat: Footer 구현 및 GNB 수정

Footer를 구현함
추가로, GNB 구현 전에도 GNB 형태를 인식할 수 있도록
GNB의 임시 높이를 20으로 수정하고 배경을 입힘

Related #3

jsh1147 committed:
Merge pull request #15 from jsh1147/feat/mentoring

레이아웃 일부 구현

jsh1147 committed:
feat: GNB 네비게이션 & 로고 구현

GNB에 속하는 네비게이션(상단, 하단)과 로고를 구현함

Related #3

chaemina committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/1-branch-account

chaemina committed:
Merge pull request #1 from Step3-kakao-tech-campus/weekly

레이아웃 구현

jsh1147 committed:
fix: 불필요 .gitkeep 제거

디렉터리 내에 파일이 생겨 더이상 불필요해진
.gitkeep 파일을 제거함

jsh1147 committed:
refactor: GNB 내비게이션 리팩토링

GNB 내비게이션을 구현할 때 사용하던
URL 정보와 내비게이션 상태를 NavState로 묶어 hook화함
추가로 포맷팅과 주석 수정을 스행함

Related #3

jsh1147 committed:
feat: GNB 계정 기능 구현

로그인 전에는 로그인 버튼이 보이고
로그인 후에는 프로필 사진과 로그아웃 버튼이
보이도록 구현함

로그인 기능이 미구현이므로
로그인 시 로컬스토리지에 token을 둔다고 가정함
(추가로 로그인/로그아웃이 가능한TEST 버튼을 둠)
API 기능이 미구현이므로 외부 이미지 경로를 사용함

Related #3

jsh1147 committed:
feat: AuthCheck 레이아웃 구현

로그인이 필요한 페이지에 접근 시
로그인 상태를 인증하는 token을 확인하도록 구현함

React.StrictMode에서는 2번 동작하므로
의도치 않은 효과가 발생하니 유의해야 함

Related #3

chaemina committed:
fix : xl size 추가

Button 컴포넌트 스타일에 xl size 추가

chaemina committed:
feat : Login page 폼 구현

Login form 에서 InputBox로 props를 전달하여, 컴포넌트의 재사용성을 높임
input의 속성 값들은 변동이 없는 값이므로LOGIN.js에 작성하여 따로
분리시켜 코드의 가독성을 높임

chaemina committed:
Merge branch 'feature/1-branch-account' of https://github.com/chaemina/Team18_FE into feature/1-branch-account

chaemina committed:
레이아웃 구현

레이아웃 구현

chaemina committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/1-branch-account

chaemina committed:
fix : useEffect 두번 발생 방지

AuthCheck에서 effect가 두번 발생함을 방지하고자, useAuthCheck 를
작성하여 hook으로 사용함

bada308 committed:
fix: tailwind 적용 안 되는 문제 해결

bada308 committed:
feat: react/prop-types 규칙 off

bada308 committed:
feat: 채팅 룸 페이지 생성

- 채팅 룸 페이지 생성
- 라우터 연결

Related #18

bada308 committed:
feat: 채팅 페이지에 사용되는 검색창 구현

- 채팅 페이지에 사용되는 검색창 구현

Related #18

chaemina committed:
refactoring : Link component로 수정

a 태그는 클릭 시 새로운 페이지를 불러오기 때문에  Link component로
수정함

bada308 committed:
feat: sendbird 관련 라이브러리 설치
- sendbird/uikit-react 설치

Related #18

bada308 committed:
Merge branch 'weekly' into feat/chat-list

bada308 committed:
Merge pull request #1 from bada308/feat/chat-list

채팅 리스트 창 컴포넌트 구현

chaemina committed:
refactoring : AccountLayout 추가

기존에 account template에서 footer 컴포넌트를 호출하여 사용했는데, 이는
요소가 넘치는 경우에 하단에 위치시키기 어려움이 있어서 accountlayout을
새로 작성하여 route로  레이아웃 적용시키도록 수정함

chaemina committed:
fix : overflow 처리

dropdown 컴포넌트의 옵션이 많아 질 경우, 컴포넌트가 스크린 범위 이상으로
커짐을 방지하고자 max-height 값을 설정하고, 제한 범위 이상일 때 y 스크롤
생성하도록 수정함

chaemina committed:
feat : 회원가입 컴포넌트 및 폼 구현

Login 폼과 동일하게 입력란은 inputBox에 props를 전달하도록 구현함
국가 선택의 경우, COUNTRY.js 상수 컴포넌트를 생성하여 국가 이름 배열을
저장하고 dropdown option의 prop으로 전달함
*멘토 선택 미 구현

chaemina committed:
로그인 페이지 폼 구현 

로그인 페이지 폼 구현

chaemina committed:
fix : 멘토님 Review PR 기반 코드 수정

4주차 코드리뷰를 바탕으로 수정함

bada308 committed:
feat: 그룹채팅 기능 구현

- sendbird uikit을 이용해 그룹채팅 기능 구현

Related #18

bada308 committed:
Merge branch 'weekly' into feat/chat-list

bada308 committed:
Merge branch 'weekly' into weekly

bada308 committed:
Update App.jsx

bada308 committed:
Merge branch 'weekly' into feat/chat-list

bada308 committed:
Merge pull request #2 from bada308/feat/chat-list

그룹채팅 기능 구현

chaemina committed:
fix : css 오류 수정

tailwind config 추가 한 "gray" color의 명칭이 기존 tailwind css와 충돌하여
"paragraph"로 명칭 변경

chaemina committed:
멘토님 Review PR 기반 코드 수정

멘토님 Review PR 기반 코드 수정

jsh1147 committed:
Merge pull request #20 from bada308/weekly

채팅 리스트 창 컴포넌트 구현

jsh1147 committed:
fix: PR merge 관련 수정

App -> Router 변경점을 수정함
더이상 App이 없으므로 App.css를 main.css로 변경함
LogIn 페이지에서 Footer가 2개 겹쳐 있는 현상을 수정함

jsh1147 committed:
Merge pull request #24 from jsh1147/feat/mentoring

PR merge 관련 수정

bada308 committed:
fix: 채팅 기능 수정

- 깨진 ui 수정

Related #18

bada308 committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feat/chat-list

bada308 committed:
Merge pull request #3 from bada308/feat/chat-list

채팅 ui 수정

bada308 committed:
Merge pull request #25 from bada308/weekly

채팅 UI 수정

jsh1147 committed:
Merge pull request #26 from Step3-kakao-tech-campus/weekly

전남대_18조_Garden_5주차 (weekly->dev)

chaemina committed:
전남대_18조_Garden_5주차 (dev->master)

전남대_18조_Garden_5주차 (dev->master)

chaemina committed:
feat : MUI 프레임 워크로 input 컴포넌트 재구현

1. 로그인, 회원가입 페이지의 경우 input 태그 보다, MUI 프레임워크에서
제공되는 TextField 컴포넌트를 사용하면 helperText나 error 속성이
지원되기 때문에 유효성 검사에 용이 할 것으로 판단하여 재구현함
2. 재사용성을 높이고 코드를 기능에 따라 분리하기 위해  유효성 검사 hook을 작성하여각 Form에 props로 정규식 전달, 검사 결과를inputbox에 전달하도록 함

Related #14
BREAKING CHANGE : MUI 프레임워크 설치 -> package 파일들 변경

jsh1147 committed:
chore: 멘토링 관련 필요 패키지 설치

멘토링 기능을 구현하는데 필요한 패키지를 설치함
추후 추가될 수 있음

Related #29

jsh1147 committed:
feat: 멘토링 api 초안 작성

멘토링 기능에 사용될 api를 axios 토대로 구현

Related #29

jsh1147 committed:
feat: 멘토링 컴포넌트 폴더 구조화

멘토링 구현에 필요한 컴포넌트를 분류하기 위해
페이지 단위로 컴포넌트 폴더를 구조화함

Related #29

kyoungsik-dev commented on src/Router.jsx:
```diff
@@ -6,33 +6,45 @@ import {
 import Dashboard from "./pages/mentoring/Dashboard.jsx";
+import ChattingRoomsPage from "./pages/chatting/ChattingRoomsPage.jsx";
+import LoginPage from "./pages/account/LoginPage";
+import SignupPage from "./pages/account/SignupPage";
```
파일 경로명에서 확장자를 일부는 표시하고 일부는 표시하지 않고 있네요.
하나로 통일하는게 좋아보입니다.
확장자를 붙이지 않는게 좀더 일반적이에요.

kyoungsik-dev commented on src/components/account/organisms/LoginForm.jsx:
```diff
@@ -0,0 +1,51 @@
+    <>
+      <form onSubmit={handleSubmit}>
+        <main className="max-w-[500px]">
+          {inputProps.map((props) => (
+            <InputBox
+              key={props.id}
+              id={props.id}
+              label={props.label}
+              type={props.type}
+              placeholder={props.placeholder}
+            />
+          ))}
```
로그인 폼의 입력칸의 경우, 아이디와 비밀번호 외 다른 칸이 추가될 가능성이 없고, 갯수도 2개 밖에 되지 않습니다.
입력칸 정보를 별도로 분리한 것은 지나친 모듈화로 보입니다.
확장성이 요구되지 않는 상황에서 확장성을 구현하여 오히려 유지보수성과 가독성을 떨어뜨린 것 같아요.

chaemina replied:
별도의 LOGIN 상수 없이 직접 값을 대입해야한다는 말씀 맞을까요?

kyoungsik-dev commented on src/layouts/GNB.jsx:
```diff
@@ -0,0 +1,186 @@
+import { useNavState } from "/src/hooks/useNavState.js";
+
+const nav = [
+  {
```
이건 별도 상수로 분리하여 모듈화하는게 좋을 것 같네요!

kyoungsik-dev commented on src/hooks/useNavState.js:
```diff
@@ -0,0 +1,11 @@
+  const currentUrl = useLocation()
+    .pathname.replace(/\d/, "")
+    .replace(/^\/+|\/+$/g, "");
+  const [selectedmainNav, setselectedMainNav] = useState("Watching");
```
```suggestion
  const [selectedMainNav, setSelectedMainNav] = useState("Watching");
```
camelCase 표기법을 제대로 지켜주세요~

kyoungsik-dev commented on src/layouts/GNB.jsx:
```diff
@@ -0,0 +1,186 @@
+
+  const handleLogOutClick = () => {
+    window.localStorage.removeItem("token");
+    setAuth(window.localStorage.getItem("token") ? true : false);
```
```suggestion
    setAuth(!!window.localStorage.getItem("token"));
```

kyoungsik-dev commented on src/layouts/GNB.jsx:
```diff
@@ -0,0 +1,186 @@
+                currentUrl.includes(val.mainUrl)
+                  ? " border-b-2 border-orange"
+                  : ""
+              }`}
```
className에 조건이 많아지니 가독성이 많이 떨어지네요.
[classnames](https://www.npmjs.com/package/classnames) 라이브러리를 써보시는건 어떨까요?
조건에 따라 클래스명을 편리하게 추가할 수 있는데요,
더 좋은 방법이 있는지도 고민해보세요~

kyoungsik-dev commented on src/layouts/GNB.jsx:
```diff
@@ -0,0 +1,186 @@
+              if (auth) window.localStorage.removeItem("token");
+              else window.localStorage.setItem("token", "Bearer 1234");
+              setAuth(window.localStorage.getItem("token") ? true : false);
+            }}
```
별도 함수로 분리하는게 좋겠죠?

kyoungsik-dev commented on src/layouts/GNB.jsx:
```diff
@@ -0,0 +1,186 @@
+      <div className="h-8 px-16 border space-x-4">
+        {nav
+          .find((val) => val.mainNav === selectedmainNav)
+          .sub.map((val) => (
```
별도 상수로 분리하는게 좋겠네요.

kyoungsik-dev commented on src/layouts/GNB.jsx:
```diff
@@ -0,0 +1,186 @@
+          .find((val) => val.mainNav === selectedmainNav)
+          .sub.map((val) => (
+            <Link
+              key={`subNav-${val.subNav}`}
```
```suggestion
              key={val.subNav}
```
이렇게만 하셔도 충분합니다.

jsh1147 committed:
fix: 멘로링 api 초안 수정

인스턴스에서 응답 세부 오류를 출력하도록 수정함
멘토링 리스트 페이지에서 사용되는 api를 수정함

Related #29

kyoungsik-dev reviewed (approved):
이번주도 고생 많으셨습니다~!

chaemina committed:
fix : dropdown 옵션으로 인해 아래요소 밀림 해결

- 최상위 div에 position relative를 부여하고, dropdown 리스트에 position
absolute를 부여함으로써 위치를 고정시킴
- 피드백으로는 select tag 사용을 추천하셔서 논의 후다시 수정 될 수 있음

chaemina committed:
refactoring : hook 이름 변경

기능을 명확히 나타내는 이름으로 변경함

chaemina committed:
fix : 로그인 페이지에서 뒤로가기 시 alert 해결

사용자 인증이 필요한 페이지 접근 시 alert 창이 뜨고 Login 페이지로
이동하는데, 이때 뒤로가기 하면 alert이 뜨는 오류가 발생함
사용자가 원하는 대로 뒤로가기가 어렵기 때문에 navigate에 replace 속성을
true로 부여하여, 로그인 페이지로 리다이렉트 하며 히스토리가 교체 되도록
함

jsh1147 committed:
chore: 패키지 설치

국기 사용을 위한 react-country-flag와
에러 처리를 위한 react-error-boundary를 설치함

jsh1147 committed:
feat: 멘토링 관련 api 기능 구현

아직 api 기능이 존재하지 않아
mock 데이터를 두고 axios로 넘기도록 구현함

Related #29

jsh1147 committed:
feat: 멘토링 관련 컴포넌트 & 훅 구현

멘토링 기능에서 사용되는
Error, Loader, 커스텀훅을 구현함

기존 Tag를 멘토&멘티에 적용 가능하도록 확장하고,
국기용 Tag를 추가함

Related #29

jsh1147 committed:
feat: 멘토링 목록 페이지 구현

멘토링 목록 페이지를 구현함
사용자의 역할에 따라 페이지에 차이가 발생하므로,
아직 로그인 상태가 없어 별도 atom을 둚
api 동작에 따라 error, loading, skeleton으로 대응함

Related #29

jsh1147 committed:
feat: GNB 기능 개선

멘토피드백을 기반으로 GNB를 개선함
- 하단Nav가 상단Nav 바로 밑에 나타나도록 수정함
- 상단 Nav 클릭 시 대표 페이지로 이동하도록 수정함
- 비로그인 상태에서 alert로 로그인 페이지에 갈 경우
  navigate에 replace 옵션을 추가하여 경로를 대체함

jsh1147 committed:
style: import 파일 경로 확장자 제거

import 파일 경로에 있는 확장자를 제거함

jsh1147 committed:
refactor: GNB 개선

내비게이션 구조 객체를 별도 파일로 분리함
그 외 추가적으로 GNB를 개선함

chaemina committed:
feat : Role check radio button 구현

checkbox 컴포넌트를 생성하여, 회원가입 폼에서 사용함
type을 radio로 지정하여 하나의 값만 선택 가능하게 함

chaemina committed:
feat : useForm hook으로 form 관리

react-hook-form에서 제공되는 useForm hook을 사용하여 로그인, 회원가입
폼을 관리하도록 수정함. 제공되는 객체를 이용하여 수월하게 양식을 관리 할
수 있으며 기존에 작성하였던 유효성 검사 훅은 제거함.
MUI text-field 와 함께 사용하기 위해  Controller를 사용함.
(inputbox는 추후 MUI를 사용하지 않는 방향으로 리팩토링 예정)

jsh1147 committed:
Merge pull request #31 from jsh1147/feat/mentoring

멘토링 목록 페이지 구현 및 기타 수정

chaemina committed:
chore : 패키지 설치

MUI 프레임워크 사용을 위한 mui/material, emotion/styled
useform 사용을 위해 react-hook-form 설치함

chaemina committed:
chore : conflict 방지 위한 패키지 설치

chaemina committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/1-branch-account

chaemina committed:
fix : conflict 방지

bada308 committed:
fix: sendbird 관련 파일 제거

- talkplus api 사용

Related #30

bada308 committed:
feat : talkplus 초기세팅

- talkplus script 추가
- user 이미지 추가
- 관련 라이브러리 설치

Related #30

bada308 committed:
docs : .gitignore에 .env 파일 추가

Related #30

bada308 committed:
feat : talkplus 관련 api 선언

- login
- getChannel
- createChannel
- getMessage 등 구현

Related #30

bada308 committed:
feat : user, channels 렌더링하는 컴포넌트 구현

- UserBox
    - 사용자 이름과 프로필 사진 렌더링
- Channel List
    - public channel list 렌더링

Related #30

bada308 committed:
feat : 문자 검사 함수 구현

Related #30

bada308 committed:
feat : 채팅 관련 UI 구현

- 보낸 메세지
- 받은 메세지
- 메세지 리스트 등을 렌더링하는 UI 구현
- 내부 로직은 구현되지 않음

Related #30

bada308 committed:
feat : chatting page 구현

- login 기능 구현

Related #30

bada308 committed:
Merge branch 'master' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feat/chat-list

bada308 committed:
Merge pull request #4 from bada308/feat/chat-list

채팅 로그인 & 채널 리스트 기능 구현

chaemina committed:
feat: jotai로 전역 상태 관리

- 모든 내부 컴포넌트에서 jotai atom을 사용 할 수 있도록 main.jsx에서 Provider로감싸줌
- token과 회원 정보 상태 저장할 atom 생성

chaemina committed:
fix : props.type 누락 수정

chaemina committed:
feat : 회원가입 전역상태관리 & 통신 구현

- 아직 로그인은 미구현
- POST : 회원가입 버튼 클릭 시 회원정보를 백앤드에 전달
- 현재 백앤드 api가 없기 때문에 instance.js에서
  console.log(config.data)로 전달 되는 데이터 값을 확인
- role이 check 되지 않은 경우, button 활성화 되지 않도록(에러메세지)

Related #14

chaemina committed:
fix : 상수 컴포넌트 수정

- 라이브러리 변화에 따라 COUNTRY 객체 code값을2자리 code로 수정함
- name camelCase로 작성

bada308 committed:
Merge branch 'weekly' into weekly

bada308 committed:
Merge pull request #32 from bada308/weekly

Talkplus 로그인 & 채널 리스트 기능 구현

chaemina committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/1-branch-account

chaemina committed:
로그인 & 회원가입 폼 수정, 전역 상태 관리 및 통신 처리 

로그인 & 회원가입 폼 수정, 전역 상태 관리 및 통신 처리

chaemina committed:
전남대_18조_Garden_6주차(weekly->dev)

전남대_18조_Garden_6주차(weekly->dev)

chaemina committed:
전남대_18조_Garden_6주차(dev->master)

전남대_18조_Garden_6주차(dev->master)
